### PR TITLE
Streams - add single call crypt functions

### PR DESCRIPTION
**Pull request by buggywhip**
Added single-call crypt functions to each of the stream ciphers using the \<cipher\>_memory() naming convention.

### Checklist
* [x] documentation is added or updated
* [x] tests are added or updated

### CODE REVIEW

File: src/stream/chacha/chacha_test.c [hunk 1]
```diff
@@ -59,6 +59,9 @@ int chacha_test(void)
    if ((err = chacha_ivctr64(&st, n + 4, sizeof(n) - 4, 1)) != CRYPT_OK)                   return err;
    if ((err = chacha_crypt(&st, (unsigned char*)pt, len, out)) != CRYPT_OK)                return err;
    if (compare_testvector(out, len, ct, sizeof(ct), "CHACHA-TV3", 1))                      return CRYPT_FAIL_TESTVECTOR;
+   /* crypt in a single call using 64-bit counter with a value of 1 */
```
Comment (sjaeckel) [inline src/stream/chacha/chacha_test.c]: Can you please add a second testcase that also hits `chacha_ivctr32()`.

**Review by sjaeckel — CHANGES_REQUESTED** [review]
LGTM :+1:

**Comment by sjaeckel** [comment]
I took the freedom to rebase everything on top of develop and clean up the duplicate commits :)

**Comment by buggywhip** [comment]
> On 10Sep, 2018, at 2:37 AM, Steffen Jaeckel <notifications@github.com> wrote:
> 
> I took the freedom to rebase everything on top of develop and clean up the duplicate commits :)
> 
Thanks.  ...but the git log graph looks funny.  ...and certainly not like it was after my rebase/squash.  ???  May I assume it is fixable?  ...how?

**Review by sjaeckel — COMMENTED** [review]
I think we should include these `xx_memory()` implementations in the single c files where there's only a single file for now!
@karel-m what do you think?
@buggywhip sorry, I didn't see that before!

**Comment by karel-m** [comment]
> I think we should include these `xx_memory()` implementations in the single c files where there's only a single file for now!

I am for keeping `xx_memory()` **always** in a separate `.c` file. It has obvious advantages (smaller code) when using static linking.

**Comment by sjaeckel** [comment]
Good point I didn't think about!
So let's keep it like that.

**Pull request merged**